curriculum/challenges/english/25-front-end-development/lab-selection-sort/680b3ef395479b0e449ecb6e.md

```diff
@@ -0,0 +1,143 @@
+---
+id: 680b3ef395479b0e449ecb6e
+title: Implement a Selection Sort Algorithm
+challengeType: 27
+dashedName: implement-a-selection-sort-algorithm
+---
+
+# --description--
+
+## Updated Description
+
+Selection sort is another popular sorting algorithm taught in most computer science courses.
+
+This algorithm works by repeatedly finding the smallest element from the unsorted portion of the list and swapping it with the first unsorted element. It begins by selecting the minimum value in the entire list and swapping it with the first element. Then it moves to the second position, finds the smallest value in the remaining unsorted elements, and swaps it with the second element. This process continues, moving through the list one element at a time, until the entire list is sorted.
+
+Selection sort results in a quadratic time complexity in the best, average, and worst case scenarios. The space complexity will be constant `O(1)` because the sorting is done in place and a constant amount of memory is being used regardless of the size of the list.
+
+## User Stories
+
+1. You should define a function named `selection_sort`.
+1. Your `selection_sort` function should have one parameter that represents the list of items.
+1. Your `selection_sort` function should take a list and sort the items in place from smallest to largest.
+1. Your `selection_sort` function should not use the built-in `sort()` method.
+
+# --hints--
+
+You should have a function named `selection_sort`.
+
+```js
+({test: () => {
+    runPython(`
+assert _Node(_code).has_function('selection_sort')
+    `)
+}})
+```
+
+Your `selection_sort` function should have one parameter.
+
+```js
+(
+    {test: () => {
+        runPython(`
+            import inspect
+            sig = inspect.signature(selection_sort)
+            assert len(sig.parameters) == 1
+        `)
+    }}
+)
+```
+
+Your `selection_sort` should not use the built-in `sort` method or any other sorting method.
```

Comment:
```suggestion
You should not use the built-in `sort()` method or `sorted()` function in your code.
```
Not very important, but I would change the hint because this test is satisfied even if you don't have a `selection_sort` function.